src/drivers/can.rs

```diff
@@ -0,0 +1,1502 @@
+
+        self.cs.set_low();
+        self.spi.write(&byte_msg)?;
+        self.cs.set_high();
+        //defmt::info!("Write instruction, Sent (MOSI): {:08b}, MISO = High-Impedance",
+        // byte_msg);
+        Ok(())
+    }
+
+    /// Changing settings, is mostly used for changing the operating mode of the
+    /// MCP2515. E.g., changing from 'Configuration' to 'Normal' mode.
+    fn change_settings(&mut self, settings: Mcp2515Settings) {
+        self.can_settings.canctrl = settings.canctrl;
+
+        let bitmask_canctrl = self.get_canctrl_mask(self.can_settings.canctrl);
+        self.write_register(MCP2515Register::CANCTRL, bitmask_canctrl);
+
+        //let canstat = self.read_register(MCP2515Register::CANSTAT,
+        // 0x00).unwrap(); defmt::info!("Bitmask for changing settings:
+        // {:08b}", bitmask_canctrl); defmt::info!("After change
+        // settings in CANSTAT: {:08b}", canstat);
+    }
+
+    /// This is just for debugging and testing the SPI, for testing read and
+    /// write for MCP2515.
+    fn transfer_test(&mut self, register: MCP2515Register) {
+        let mut read_buf: [u8; 3] = [0; 3];
+        let write_buf: [u8; 3] = [InstructionCommand::Read as u8, register as u8, 0x00];
+        self.cs.set_low();
+        self.spi.transfer(&mut read_buf, &write_buf);
+        self.cs.set_high();
+        defmt::info!("Sent: {:08b}, Received: {:08b}", write_buf, read_buf);
+    }
+
+    /// Loopback test, that sets the mode to loopback, and LOAD TX and READ RX
+    /// buffer.
+    /// -------------------------------------------------------------------------*
+    /// (1). Change mode by writing to CANCTRL register with the settings
+    /// bitmask. (2). Create a new CAN frame and convert to byte array.
+    /// (3). Perform 'Load TX buffer' instruction, loading the CAN byte frame.
+    /// (4). Initiate transmission by setting the TXREQ bit (`TXBnCTRL`[3])
+    /// for each buffer. (5). Poll the target RX buffer, checking if message
+    /// has been received. Then read RX!
+    /// -------------------------------------------------------------------------*    
+    /// Example format of byte message: ...
+    /// ... let `byte_msg`: [u8; 3] = [`InstructionCommand::Write` as u8,
+    /// `MCP2515Register::TXB0CTRL` as u8, data];
+    pub fn loopback_test(&mut self, can_msg: CanMessage) {
+        defmt::info!("Loopback Test:");
+        let mut can_settings = Mcp2515Settings::default();
+        let canctrl_settings =
+            SettingsCanCtrl::new(OperationTypes::Loopback, false, CLKPRE::DIV1, false, false);
+        can_settings.canctrl = canctrl_settings;
+
+        self.change_settings(can_settings);
+
+        //self.read_status();
+        self.load_tx_buffer(TXBN::TXB0, can_msg);
+        //self.poll_rx(RXBN::RXB0);
+        //self.rx_status();
+    }
+
+    /// The `poll_rx` method was for reading the RX buffers for a loopback test
+    /// without interrupts.
+    fn poll_rx(&mut self, rx: &RXBN) {
```

Comment:
I think that this could be a good thing to include, but I would rename it to 
`blocking_rx`

Reply:
Renamed as suggested!